<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyAgreeDlg</class>
 <widget class="QDialog" name="KeyAgreeDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>564</width>
    <height>482</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Key Agreement</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="3" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="mSecretClearBtn">
       <property name="text">
        <string>Secret Clear</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="mACalcBtn">
       <property name="text">
        <string>A Calculate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="mBCalcBtn">
       <property name="text">
        <string>B Calculate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="mCloseBtn">
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Secret Key</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QPlainTextEdit" name="mSecretKeyText"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QTabWidget" name="mTabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>DH</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QGroupBox" name="mDHGroup">
         <property name="title">
          <string>DH</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="6" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>A PublicKey</string>
            </property>
           </widget>
          </item>
          <item row="6" column="5">
           <widget class="QPushButton" name="mAGenDHKeyBtn">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>A GenDHKey</string>
            </property>
           </widget>
          </item>
          <item row="8" column="5">
           <widget class="QPushButton" name="mBDHPriBtn">
            <property name="text">
             <string>B GenDHPri</string>
            </property>
           </widget>
          </item>
          <item row="1" column="5">
           <widget class="QComboBox" name="mGCombo"/>
          </item>
          <item row="6" column="1" colspan="3">
           <widget class="QLineEdit" name="mAPublicKeyText"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Length</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1" colspan="3">
           <widget class="QLineEdit" name="mBPrivateKeyText"/>
          </item>
          <item row="4" column="0" colspan="6">
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="5" column="5">
           <widget class="QPushButton" name="mADHPriBtn">
            <property name="text">
             <string>A GenDHPri</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>B PublicKey</string>
            </property>
           </widget>
          </item>
          <item row="9" column="5">
           <widget class="QPushButton" name="mBGenDHKeyBtn">
            <property name="text">
             <string>B GenDHKey</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1" colspan="3">
           <widget class="QLineEdit" name="mAPrivateKeyText"/>
          </item>
          <item row="3" column="0" colspan="6">
           <widget class="QPlainTextEdit" name="mPText"/>
          </item>
          <item row="1" column="3">
           <widget class="QLineEdit" name="mGText">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="2" column="5">
           <widget class="QPushButton" name="mGenParamBtn">
            <property name="text">
             <string>GenParam</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>A PrivateKey</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>p value</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1" colspan="3">
           <widget class="QLineEdit" name="mBPublicKeyText"/>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>B PrivateKey</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>g value</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="mLengthText"/>
          </item>
          <item row="7" column="0" colspan="6">
           <widget class="Line" name="line_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>ECDH</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QGroupBox" name="mECDHGroup">
         <property name="title">
          <string>ECDH</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="3" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>A PublicKey</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QComboBox" name="mECDHParamCombo"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>A PrivateKey</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="5">
           <widget class="QLineEdit" name="mAECDHPriKeyText"/>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>B PublicKey</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Parameter</string>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <widget class="QPushButton" name="mAGenPriKeyBtn">
            <property name="text">
             <string>A GenPriKey</string>
            </property>
           </widget>
          </item>
          <item row="4" column="5">
           <widget class="QPushButton" name="mAFindPriKeyBtn">
            <property name="text">
             <string>A Find PriKey</string>
            </property>
           </widget>
          </item>
          <item row="10" column="4">
           <widget class="QPushButton" name="mBGenPubKeyBtn">
            <property name="text">
             <string>B GenPubKey</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="5">
           <widget class="QLineEdit" name="mAECDHPubKeyText"/>
          </item>
          <item row="10" column="3">
           <widget class="QPushButton" name="mBGenPriKeyBtn">
            <property name="text">
             <string>B GenPriKey</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1" colspan="5">
           <widget class="QLineEdit" name="mBECDHPubKeyText"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>B PrivateKey</string>
            </property>
           </widget>
          </item>
          <item row="4" column="4">
           <widget class="QPushButton" name="mAGenPubKeyBtn">
            <property name="text">
             <string>A GenPubKey</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1" colspan="5">
           <widget class="QLineEdit" name="mBECDHPriKeyText"/>
          </item>
          <item row="10" column="5">
           <widget class="QPushButton" name="mBFindPriKeyBtn">
            <property name="text">
             <string>B Find PriKey</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="6">
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="6">
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
